lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,276 @@
+
+        if action not in ['enroll', 'unenroll']:
+            error_msg = "action must be enroll or unenroll"
+        elif not isinstance(courses, (list, tuple)):
+            error_msg = "courses_id must be a list of courses_id "
+        elif not isinstance(users, (list, tuple)):
+            error_msg =  "users must be a list of identifiers(username or email)"
+        else:
+            error_msg = None
+
+        if error_msg:
+            return Response(
+                data={"error_message": error_msg},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            result = self.__enroll_unenrol_users_for_courses(
+                courses, users, request,
+                action, email_students, auto_enroll
+            )
+
+            return Response(
+                data={
+                    'action': action,
+                    'courses': result,
+                    'email_students':email_students,
+                    'auto_enroll': auto_enroll,
+                },
+                status=status.HTTP_200_OK
+            )
+        except  Exception as e:
+            log.error(e.message)
+            return Response(
+                data={"error_message": e.message},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    def __enroll_unenrol_users_for_courses(
+            self, courses_id, identifiers, request,
+            action='enroll', email_students=False, auto_enroll=False
+    ):
+        courses = {}
+        for course_id in courses_id:
+            results = self._enroll_unenrol_users_for_course(
+                course_id, identifiers, request,
+                action, email_students, auto_enroll
+            )
+            courses[course_id] = self._make_result_dict_for_course(
+                 results, action, auto_enroll
+            )
+        return courses
+
+    def _enroll_unenrol_users_for_course(
+            self, course_id, identifiers, request,
+            action='enroll', email_students=False, auto_enroll=False
+    ):
+        result_list = []
+        for identifier in identifiers:
+            try:
+                course_id_obj = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+                course = get_course_by_id(course_id_obj)
+                if email_students:
+                    email_params = get_email_params(course, auto_enroll, secure=request.is_secure())
+                else:
+                    email_params = None
+
+                try:
+                    user = get_student_from_identifier(identifier)
+                    email = user.email
+                    language = get_user_email_language(user)
+                except User.DoesNotExist as e:
+                    if auto_enroll and action == 'enroll':
+                        email = identifier
+                        language = None
+                    else:
+                        raise e
```

Comment:
Specifying exception is not needed if you what to raise the same one which was caught by the try-except block.